### PR TITLE
sass 1.22.10 (new formula)

### PR DESCRIPTION
**Pull request by JarryShaw**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Pure JavaScript implementation of SASS, other than `libsass` & `sassc`.

### CODE REVIEW

**Comment by fxcoudert** [comment]
It's currently the top #415 formula installed in Homebrew, with 4234 installs per month, so sure! 🎈

File: Formula/sass.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+require "language/node"
+
+class Sass < Formula
+  desc "Pure JavaScript implementation of SASS"
```
Comment (fxcoudert) [inline Formula/sass.rb]: Having `sass` describe itself as an “implementation of SASS” is not great. Can you think of something better?

File: Formula/sass.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    (testpath/"test.scss").write <<~EOS
+      div {
+        img {
+          border: 0px;
+        }
+      }
+    EOS
+
+    assert_equal "div img{border:0px}",
+    shell_output("#{bin}/sass --style=compressed test.scss").strip
```
Comment (fxcoudert) [inline Formula/sass.rb]:
```
output = shell_output("#{bin}/sass --style=compressed test.scss").strip
assert_equal "div img{border:0px}", output
```

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by fxcoudert** [comment]
Please squash all changes into a single commit, with message `sass 1.22.10 (new formula)`

**Comment by chdiza** [comment]
I don't think this should be called `sass`.  The "main" implementation of sass is [the dart one](github.com/sass/dart-sass).  I don't know why a formula called `sass` would do an `npm` install of some kind of wrapper for the real sass.  I would expect a formula called `sass` to install the dart version, which doesn't require `npm` or `node`.

**Comment by JarryShaw** [comment]
If you check [the Sass website](https://sass-lang.com/dart-sass), "Dart Sass is also distributed as the pure JavaScript `sass` package on `npm`", whilst **the dart one** is more likely a `dart` library (managed by `pub`) rather than a standalone CLI tool.

**Comment by chdiza** [comment]
If you look on that website, you'll also find that the dart one is in fact released as a standalone tool.  It can be found on the sass/dart-sass repo under "releases".  I've been using it for a long time.  It does not require `node` or `npm`.  You will also find that sass-lang.com/install says:

"If you use Node.js, you can also install Sass using npm by running
npm install -g sass
However, please note that this will install the pure JavaScript implementation of Sass, which runs somewhat slower than the other options listed here."

So this PR proposes a slower version of the real tool while also requiring an additional unnecessary dependency.

On top of that, I thought that HB discouraged formulas that are really nothing but `npm/gem/pip install foo`, but perhaps they've changed their mind on that.

**Comment by SMillerDev** [comment]
Without knowing anything about sass, I'd say a preferred install by upstream with less dependencies should be what we're adding to homebrew. Even if they ship other implementations. And the project already has a formula that can just be copied over I guess.

**Comment by JarryShaw** [comment]
So, actually as @SMillerDev said, the Sass project itself had provided a formula `sass/sass/sass` from their own Tap. This formula requires `dart` when building but not at runtime, which probably seems much more friendly.

Therefore, I would suggest renaming this formula as `node-sass` (or let's forget about it) and copy the `sass/sass/sass` as the *real* Sass formula.

**Comment by SMillerDev** [comment]
Which sass is this now? JS or Dart?

**Comment by chdiza** [comment]
It's still JS.

**Comment by JarryShaw** [comment]
Alright, let's copy over the dart formula to Homebrew core, shall we? 

Beside that, either let's rename the formula I'm comitting as `node-sass` and open a new pr for it; or let's just close the case and forget about it. Decision on you guys :)

**Comment by SMillerDev** [comment]
I'm fine with either. If you only need one implementation I'd stick with the Dart one though.

**Comment by JarryShaw** [comment]
Closing this pull request as I just filed two separate pull requests for both node (#47439) and dart (#47438) implementation of SASS.

**Pull request closed without merging**